### PR TITLE
[WIP] feature 1264

### PR DESCRIPTION
**Pull request by brouillette**
Search: When changing parameters, refresh the Count automatically instead of requiring the user to click "Get Data"

### CODE REVIEW

**Review by davidlougheed — CHANGES_REQUESTED** [review]
in general i've suggested a couple directions to avoid prop-drilling with the dispatch function

**Review by brouillette — COMMENTED** [review]
thanks
much cleaner now

File: src/js/components/Search/SelectOption.js [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
+const SelectOption = ({ id, isChecked, options, optionalDispatch}) => {
   const { t } = useTranslation();
   const dispatch = useDispatch();
 
   const [value, setValue] = useState(options[0]);
 
+  const isFirstRun = useRef(true);
   useEffect(() => {
     if (isChecked) {
       dispatch(addQueryParam({ id, value }));
```
Comment (v-rocheleau) [inline src/js/components/Search/SelectOption.js:17]: If the user un-checks this option, won't the query param be out of sync?

File: src/js/components/Search/SelectOption.js [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
     }
+
+    // to prevent everything between the lines '--'
+    // at component initialization
+    if (isFirstRun.current) {
+      isFirstRun.current = false;
+      return;
+    }
+    // --
+    if (optionalDispatch != undefined) {
```
Comment (v-rocheleau) [inline src/js/components/Search/SelectOption.js:27]: We could use the truthy form `if (optionalDispatch) {...}` to handle unexpected types

Comment (davidlougheed) [reply src/js/components/Search/SelectOption.js:27]: also fwiw `!=` should never be used in JS anyway, always `!==`

File: src/js/components/Search/SelectOption.js (hunk 1 above)
Comment (brouillette) [reply src/js/components/Search/SelectOption.js:17]: Nope
This 'first-run' is only triggered upon first page and component load

**Review by v-rocheleau — APPROVED** [review]

**Review by davidlougheed — APPROVED** [review]

**Pull request merged**